crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3706,16 +3707,44 @@ impl<'db> TypeInferenceBuilder<'db> {
             .member(db, &attr.id)
             .unwrap_with_diagnostic(|lookup_error| match lookup_error {
                 LookupError::Unbound => {
+                    let bound_on_instance = match value_type {
+                    Type::ClassLiteral(ClassLiteralType { class }) => {
+                        !class.instance_member(self.db(), attr).0.is_unbound()
```

Comment:
there's a `let db = self.db();` assignment a few lines higher so that this method doesn't have to have lots of verbose `self.db()` calls everywhere

```suggestion
                        !class.instance_member(db, attr).0.is_unbound()
```